hud/src/commonMain/kotlin/HUD.kt

```diff
@@ -117,12 +115,11 @@ interface HUD {
      * hides view after block finished.
      * @param block The block to execute with hud visible
      */
-    fun presentDuring(animated: Boolean = true, block: suspend () -> Unit): HUD = apply {
-        present(animated) {
-            GlobalScope.launch(MainQueueDispatcher) {
-                block()
-                dismiss(animated)
-            }
+    suspend fun presentDuring(animated: Boolean = true, block: suspend () -> Unit): HUD = apply {
```

Comment:
Instead of returning HUD (which is sort of useless since its already dismissed by the time this method returns), can we give it a generic return type (as determined by `block`)? I noticed this is useful when the block needs to do things that can only be done after the HUD has been dismissed (e.g. show an Alert)